src/history/hs.c

```diff
@@ -424,6 +427,17 @@ __hs_insert_record_with_btree(WT_SESSION_IMPL *session, WT_CURSOR *cursor, const
     WT_ERR(__wt_row_modify(cbt, &cursor->key, NULL, hs_upd, WT_UPDATE_INVALID, true));
 
 err:
+    if (ret == 0 && upd->start_ts == WT_TS_NONE) {
```

Comment:
Can you add a comment to explain how the non-timestamped tables are handled? The updates on those tables have the `start_ts` as `WT_TS_NONE`. Explain the search the logic of how we do will avoid the deletion of the history store values.

Reply:
12bed73